internal/support/notifications/sending_service.go

```diff
@@ -141,3 +135,87 @@ func handleFailedTransmission(t models.Transmission) {
 		}
 	}
 }
+
+func deduceAuth(s SmtpInfo) (mail.Auth, error) {
+	if s.Username == "" && s.Password == "" {
+		return nil, errors.New("Notifications: Expecting username")
+	}
+	if s.Username != "" && s.Password == "" {
+		return nil, nil
+	}
+	if s.Username == "" && s.Password != "" {
+		return nil, errors.New("Notifications: Expecting username")
+	}
+	return mail.PlainAuth("", s.Username, s.Password, s.Host), nil
```

Comment:
I have a problem with this function because it isn't fully backward compatible. The previous implementation used `s.Sender` as part of the mail.PlainAuth call and not `s.Username`. If `s.Username` is blank, ideally this should fall back to `s.Sender` since that's what the Edinburgh release uses.

Consider adding the following function to the `SmtpInfo` type and then using that in the checks and `PlainAuth` above.
```
func(s SmtpInfo) CheckUsername() string {
	if s.Username != "" {
		return s.Username
	}
	return s.Sender
}
```

Reply:
Good point.

@rudderfree , can you also add comments stating that the function exists for backwards compatibility, so that later on we are not confused as to the intent of the logic?

Reply:
Yes, agreed. Added the function and the comment. Thx!